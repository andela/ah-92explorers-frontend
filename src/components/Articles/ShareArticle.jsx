import React from 'react';
import { APP_URL_FRONTEND } from '../../utils/constants';

export class ShareArticle extends React.Component {
  shareSocial = (provider) => {
    const { fetched, article } = this.props;
    const url = `${APP_URL_FRONTEND}/article/${article.slug}`;
    const tweet = `"${article.title}" - by ${fetched && article.author.username} @ ${APP_URL_FRONTEND}/article/${article.slug}`;
    switch (provider) {
      case 'facebook':
        window.open(`https://www.facebook.com/sharer/sharer.php?u=${url}`, '_blank');
        break;
      case 'twitter':
        window.open(`https://twitter.com/intent/tweet?text=${tweet}&hashtags=92-Explorers`, '_blank');
        break;
      case 'email':
        window.open(`mailto:?subject=${article.title}&body=Checkout this article "${article.title}" by ${fetched && article.author.username} on Authors Haven. ${'\n\n'} @ ${url}`);
        break;
      default:
    }
  };

  render() {
    return (
      <div className="socialNet">
        <img
          src={require('../../assets/icons/facebook-logo-in-circular-button-outlined-social-symbol.svg')}
          className="bodyIcons shareIcons fb"
          alt="share to facebook"
          onClick={() => this.shareSocial('facebook')}
        />
        <img
          src={require('../../assets/icons/twitter-circular-button.svg')}
          className="bodyIcons shareIcons tw"
          alt="share to twitter"
          onClick={() => this.shareSocial('twitter')}
        />
        <img
          src={require('../../assets/icons/email.svg')}
          className="bodyIcons shareIcons gp"
          alt="share via email"
          onClick={() => this.shareSocial('email')}
        />
      </div>
    );
  }
}

export default ShareArticle;
