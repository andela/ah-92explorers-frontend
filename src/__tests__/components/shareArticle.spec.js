import '@babel/polyfill';
import React from 'react';
import { shallow, mount} from 'enzyme';
import { ShareArticle } from '../../components/Articles/ShareArticle.jsx';

describe('<ShareArticle />', () => {
  let wrapper;

  const props = {
    fetched: '',
    article: '',
  };

  beforeEach(() => {
    wrapper = shallow(<ShareArticle {...props} />);
  });

  it('should render Login component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shoudl handle on comment', () => {
    const wrapper = shallow(
      <ShareArticle {...props} />,
    );
    const component = wrapper.instance();
    component.shareSocial();
    expect(component).toBeDefined();
  });

  it('hould simulate on click facebook icon', async () => {
    const btn = wrapper.find('.fb');
    btn.simulate('click', { preventDefault: jest.fn() });
    expect(btn.length).toBe(1);
  });

  it('hould simulate on click twitter icon', async () => {
    const btn = wrapper.find('.tw');
    btn.simulate('click', { preventDefault: jest.fn() });
    expect(btn.length).toBe(1);
  });

  it('hould simulate on click mail icon', async () => {
    const btn = wrapper.find('.gp');
    btn.simulate('click', { preventDefault: jest.fn() });
    expect(btn.length).toBe(1);
  });
});
