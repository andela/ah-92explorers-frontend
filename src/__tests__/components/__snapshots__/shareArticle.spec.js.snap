// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareArticle /> should render Login component 1`] = `ShallowWrapper {}`;
